### DISCUSSION TITLE
TP Link Control

### DISCUSSION THREAD

**Discussion by MrRexRuthless** [UnchartedBull/OctoDash#2643] [Q&A]
Is it possible to use a custom command to turn lights on and off using the TPLink-Plug-in.  I can do it from OctoPrint Web interface but would like to do it from the touchscreen if possible.

**Comment by jneilliii** [comment]
Yes, the plugin supports gcode triggers for turning on/off the plug. Unfortunately, I don't have a toggle option. Basically enable the gcode option on the individual plug's settings and then add a custom action that runs `@TPLINKON 192.168.0.2` with the IP of your plug or `@TPLINKOFF 192.168.0.2`.

**Reply by nowkillkennys** [reply]
I keep getting 409 error

**Reply by jjfollowell** [reply]
Same here, 409 error. Not sure how GCODE would work if the printer is off anyway.
I can control it from Octoprint, so it's defiantly working, but the command does seem to get through for turning the printer on.
I just tried with a different plug, with the printer already on, and the newpluf turns on just fine with @TPLINKON as described above, but only if the printer is on. So will be useful for turning the enclosure lights on and off, and could turn the printer off, but won't be able to turn it on in this configuration.
Is there a way to configure it so it can turn the printer itself on?

**Reply by jneilliii** [reply]
yeah, if you don't have backpowering the gcode command wouldn't work, but there are built in [!POWERON] and [!POWEROFF] custom actions that do work, that's what I use. eventually I will add toggle support and it can be simplified to a single button.

**Reply by jjfollowell** [reply]
Can't get that to work either - ah well, perhaps I need to allow back powering. probably simpler, as long as it doesn't cause low power warnings again.

**Reply by jneilliii** [reply]
That's strange. And the plugin is enabled in OctoDash's settings?

**Reply by jneilliii** [reply]
From the companion plugin you can set it like this. Mine has /4 because I'm using the strip. It just needs to match the IP field in OctoPrint's TPLink-Smartplug settings. 

![image](https://user-images.githubusercontent.com/5249455/185262194-8ae0ee5c-9ce0-424a-b12a-f47629fc470d.png)

**Reply by jjfollowell** [reply]
tried, but so far only got it working when back powering - so it works for now

it's been great by the way. Only thing missing for me is an ETA for when it will finish

keep up the great work
